📖 update dev docs for caph (#1241)
syself/cluster-api-provider-hetzner: docs/developers/development.md

# Developing Cluster API Provider Hetzner

Developing our provider is quite easy. Please follow the steps mentioned below: 
1. You need to install some base requirements. 

> 2. You need to follow the [preparation document](docs/topics/preparation.md) to setup everything related to Hetzner. 

Reviewer: setup -> set up

> 2. You need to follow the [quickstart document](docs/topics/quickstart.md) to set up everything related to Hetzner. Third, you need to configure your tilt set-up. After having done those three steps, you can start developing with the local Kind cluster and the Tilt UI to create one of the different workload clusters that are already pre-configured. 

Reviewer: 1. I think it is not (yet) quickstart.md. 
2. set-up -> setup

## Install Base requirements

> In order to develop with Tilt, there are a few requirements. You can use the following command to check whether the versions of the tools are up-to-date and to install ones that are missing: `make all-tools`

Reviewer: ```suggestion
To develop with Tilt, there are a few requirements. You can use the command `make all-tools` to check whether the versions of the tools are up to date and to install the ones that are missing.
```

- tilt (required)
- hcloud

## Preparing Hetzner project

For more information, please see [here](/docs/topics/preparation.md).

## Setting Tilt up

> You need to create a `.envrc` file and specify the values you need. After the `.envrc` is loaded, you'll have to use `direnv allow` to load the environment variables in your current shell session.

Reviewer: avoid "you'll" or something like that

$ make delete-mgt-cluster
```

> To delete the registry, use: `make delete-registry` or use `make delete-mgt-cluster-registry` to delete both management cluster and associated registry.

Reviewer: ```suggestion
To delete the registry, use `make delete-registry`. Use `make delete-mgt-cluster-registry` to delete both management cluster and associated registry.
```

> To delete the registry, use: `make delete-registry` or use `make delete-mgt-cluster-registry` to delete both management cluster and associated registry as well.

Reviewer: ```suggestion
To delete the registry, use: `make delete-registry` or use `make delete-mgt-cluster-registry` to delete both management cluster and associated registry.
```

```
export HETZNER_SSH_PRIV=$(cat ~/.ssh/cluster | base64 -w0)
```

### Creating new user in Robot 

> If you want to create new user in Robot then you can create a new user by clicking on the `Create User` button in the Hetzner Robot console. Once you create the new user, an user ID will be provided to you via email from hetzner robot and the password will be the same that you used while creating the user.

Reviewer: ```suggestion
To create new user in Robot, click on the `Create User` button in the Hetzner Robot console. Once you create the new user, a user ID will be provided to you via email from Hetzner Robot. The password will be the same that you used while creating the user.
```
